pkg/splunk/enterprise/clustermaster.go

```diff
@@ -174,7 +185,12 @@ func ApplyClusterMaster(client splcommon.ControllerClient, cr *enterprisev1.Clus
 		}
 
 		if cr.Status.BundlePushTracker.NeedToPushMasterApps == false {
-			result.Requeue = false
+			// requeue the reconcile after polling interval if we had set the flag before
+			if cr.Status.AppContext.AppInfoStatus.NeedToCheckAfterPollInterval == true {
+				result.RequeueAfter = time.Second * time.Duration(cr.Spec.AppFrameworkConfig.AppsRepoPollInterval)
+			} else {
+				result.Requeue = false
+			}
```

Comment:
What happens, if there is a reconcile event for some other reason, before the expiry? I think, in that case, we are pre-maturely resetting it? I think we need additional intelligence to pick the exact time for the next to reconcile events to accommodate any reconcile events for other non-app framewokr reasons.